tools/yulPhaser/GeneticAlgorithms.cpp

```diff
@@ -48,3 +50,28 @@ void RandomAlgorithm::runNextRound()
 			m_options.maxChromosomeLength
 		);
 }
+
+void GenerationalElitistWithExclusivePools::runNextRound()
+{
+	double elitePoolSize = 1.0 - (m_options.mutationPoolSize + m_options.crossoverPoolSize);
+	RangeSelection elite(0.0, elitePoolSize);
+
+	m_population =
+		m_population.select(elite) +
+		m_population.select(elite).mutate(
+			RandomSelection(m_options.mutationPoolSize / elitePoolSize),
```

Comment:
This number is `mps / (1 - (mps + cps))` - can you explain?

Reply:
I want the number of individuals in the mutation pool to be `mutationPoolSize * m_population.size()`.

The selection size is relative to the population I'm applying it to - which is `m_population.select(elite)` - not `m_population`.

The size of the elite is `elitePoolSize * m_population.size()` so I must set the selection size to `mutationPoolSize / elitePoolSize` to get `mutationPoolSize / elitePoolSize * elitePoolSize * m_population.size()` individuals.